Feature/ssl conf

richseviora opened the pull request:
This PR makes the following changes, fixes #205.

- Adds `ssl_cert_dir` to the Apache 2 webserver output.
- Updates the Passenger and Upstream versions of the Apache 2 template to use `ssl_cert_dir` for the SSL certificate paths.
- Adds the `X-Forwarded-Proto https` header to the Passenger template.

ajgon commented on .ruby-gemset:
```diff
@@ -1 +1 @@
-opsworks_ruby
+-global
```
Why this change?

richseviora replied:
Oops! Fixed!

ajgon commented on libraries/drivers_webserver_apache2.rb:
```diff
@@ -26,7 +26,8 @@ def self.passenger_supported?
         output = super.merge(
-          log_dir: node['deploy'][app['shortname']][driver_type]['log_dir'] || "/var/log/#{service_name}"
+          log_dir: node['deploy'][app['shortname']][driver_type]['log_dir'] || "/var/log/#{service_name}",
+          ssl_cert_dir: ssl_cert_dir
```
There are two conflicting ideas here. `output` is supposed to contain only options configurable by user. With this approach, you "hardcode" `ssl_cert_dir` despite allowing it to be configured in line 11 of this file. So I suggest one of two options:

* Make directory configurable
  * It basically means altering this line to: `ssl_cert_dir: node['deploy'][app['shortname']][driver_type]['ssl_cert_dir'] || default_ssl_cert_dir` and renaming the method to `default_ssl_cert_dir` to avoid confusion.
  * Please, don't forget to add this option to documentation if you choose this path.
* Keep the original concept:
  * remove `ssl_cert_dir` from filter options in line 11
  * remove this line
  * in file `libraries/drivers_webserver_base.rb` in `add_appserver_config` method add another key-value pair to `opts` variable:
    ```ruby
    opts = { application: app, deploy_dir: deploy_dir(app), out: out, conf_dir: conf_dir, adapter: adapter, name: a.adapter, deploy_env: deploy_env, appserver_config: a.webserver_config_params, ssl_cert_dir: ssl_cert_dir }
    ```
  * and last, but not least - alter the templates with `@ssl_cert_dir` instead of `out[:ssl_cert_dir)`

richseviora replied:
Second option sounds easier; I can't think of any immediate reason to make the SSL dir configurable.

richseviora replied:
So that's been done + I'd like to capture the architectural direction about `output` somewhere. Any recommendations? I was thinking perhaps `Drivers::DSL::Output`?

richseviora commented:
I've addressed the PR feedback as follows:

- Moved `ssl_cert_dir` to the app server template config options and adjusted the Apache templates accordingly.
- Applied the same change to the Nginx template.
- Squashed commits & reworded, though I'm going to add more content in the commit body now.

ajgon commented on .gitignore:
```diff
@@ -23,3 +23,8 @@ docs/build/*
+# Ignore Jetbrains config
+.idea
+# Ignore VS Code
+.vscode
```
Please, don't pollute project `.gitignore` with this files, and use [global](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) or [project local](https://help.github.com/en/articles/ignoring-files#explicit-repository-excludes) ignore file instead.

ajgon commented on .ruby-gemset:
```diff
@@ -1 +1 @@
-opsworks_ruby
+-opsworks_ruby
```
I assume this `-` (minus) sign got there by accident? 😄

richseviora commented:
Updated :)
